modules/apps/message-boards/message-boards-web/src/main/java/com/liferay/message/boards/web/internal/portlet/action/SelectCategoryMVCRenderCommand.java

```diff
@@ -36,12 +38,25 @@ public class SelectCategoryMVCRenderCommand implements MVCRenderCommand {
 	public String render(
 		RenderRequest renderRequest, RenderResponse renderResponse) {
 
+		MBCategory category = null;
 		long categoryId = ParamUtil.getLong(renderRequest, "mbCategoryId");
+		String mbCategoryExternalReferenceCode = ParamUtil.getString(
+			renderRequest, "mbCategoryExternalReferenceCode");
 
 		if (categoryId != MBCategoryConstants.DEFAULT_PARENT_CATEGORY_ID) {
-			MBCategory category = _mbCategoryLocalService.fetchMBCategory(
-				categoryId);
+			category = _mbCategoryLocalService.fetchMBCategory(categoryId);
+		}
+		else if (!Validator.isBlank(mbCategoryExternalReferenceCode)) {
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)renderRequest.getAttribute(WebKeys.THEME_DISPLAY);
+
+			category =
+				_mbCategoryLocalService.fetchMBCategoryByExternalReferenceCode(
+					mbCategoryExternalReferenceCode,
+					themeDisplay.getScopeGroupId());
+		}
```

Comment:
This is starting to look a bit messy. Can you extract this logic into its own `_getCategory` method?
```
MBCategory category = _getCategory(renderRequest);

if (category != null) {
    ...
}
```

Reply:
Done. See last commit [574c892](https://github.com/liferay-content-management/liferay-portal/pull/5927/commits/574c892318435a0934d4a9a824b8ecb6d522ebb8)